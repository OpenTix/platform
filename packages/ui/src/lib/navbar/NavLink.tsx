import React, { FC } from 'react';
import { Link, LinkProps } from 'react-router-dom';
import styled from 'styled-components';

const StyledNavLink = styled(Link)`
	color: #333;
	text-decoration: none;
	margin: 0 8px;
	font-size: 16px;

	&:hover {
		text-decoration: underline;
	}
`;

export interface NavLinkProps extends LinkProps {
	children: React.ReactNode;
}

export const NavLink: FC<NavLinkProps> = ({ children, ...props }) => {
	return <StyledNavLink {...props}>{children}</StyledNavLink>;
};
