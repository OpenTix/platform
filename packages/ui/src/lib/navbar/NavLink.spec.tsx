import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { NavLink } from './NavLink';

// Adjust path as needed

describe('NavLink', () => {
	it('renders the link and children', () => {
		render(
			<MemoryRouter>
				<NavLink to="/test-path">Test Link</NavLink>
			</MemoryRouter>
		);
		// The link text should be in the document
		expect(screen.getByText('Test Link')).toBeInTheDocument();
	});

	it('has the correct "to" attribute', () => {
		render(
			<MemoryRouter>
				<NavLink to="/another-path">Another Link</NavLink>
			</MemoryRouter>
		);
		const link = screen.getByText('Another Link') as HTMLAnchorElement;
		expect(link.getAttribute('href')).toBe('/another-path');
	});

	it('matches snapshot', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<NavLink to="/">Snapshot Test</NavLink>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
