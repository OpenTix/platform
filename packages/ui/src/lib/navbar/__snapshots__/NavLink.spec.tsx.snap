// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="NavLink__StyledNavLink-sc-8cdbnb-0 bcjHjZ"
    href="/"
  >
    Snapshot Test
  </a>
</DocumentFragment>
`;
